src/deserialize.cpp

```diff
@@ -0,0 +1,290 @@
+#include "sparrow_ipc/deserialize.hpp"
+
+#include <sparrow/types/data_type.hpp>
+
+#include "sparrow_ipc/deserialize_fixedsizebinary_array.hpp"
+#include "sparrow_ipc/deserialize_primitive_array.hpp"
+#include "sparrow_ipc/deserialize_variable_size_binary_array.hpp"
+#include "sparrow_ipc/magic_values.hpp"
+#include "sparrow_ipc/metadata.hpp"
+
+namespace sparrow_ipc
+{
+    void deserialize_schema_message(
+        std::span<const uint8_t> data,
+        size_t& current_offset,
+        std::optional<std::string>& name,
+        std::optional<std::vector<sparrow::metadata_pair>>& metadata
+    )
+    {
+        const uint32_t schema_meta_len = *(reinterpret_cast<const uint32_t*>(data.data() + current_offset));
+        current_offset += sizeof(uint32_t);
+        const auto schema_message = org::apache::arrow::flatbuf::GetMessage(data.data() + current_offset);
+        if (schema_message->header_type() != org::apache::arrow::flatbuf::MessageHeader::Schema)
+        {
+            throw std::runtime_error("Expected Schema message at the start of the buffer.");
+        }
+        const auto flatbuffer_schema = static_cast<const org::apache::arrow::flatbuf::Schema*>(
+            schema_message->header()
+        );
+        const auto fields = flatbuffer_schema->fields();
+        if (fields->size() != 1)
```

Comment:
I suppose you plan to remove this limitation later? Otherwise you'll fail on most real-world data.